### DISCUSSION TITLE
Is there a easy way to (re)build a single test?

### DISCUSSION THREAD

**Discussion by amneetb** [IBAMR/IBAMR#1581] [Q&A] [answered]
Currently I am doing `make -jX tests` to recompile a single test, which takes a while and builds all tests.

**Comment by abarret** [comment, marked as answer]
If you are using the CMake build system, then you just need the target for the specific test. Directories have a specific target, e.g. `make tests-<folder name>`. To build a single test, you can do `make tests-<folder name>_<test name>` (for example `tests/adv_diff/adv_diff_02_3d`), you can do `make tests-adv_diff_adv_diff_02_3d`

**Reply by amneetb** [reply]
Nice! 

(I'm using autotools and I just tried with that, and it doesn't work with it.)
